indexer/editable_map_object.cpp

```diff
@@ -83,8 +170,43 @@ void EditableMapObject::SetName(StringUtf8Multilang const & name) { m_name = nam
 void EditableMapObject::SetName(string name, int8_t langCode)
 {
   strings::Trim(name);
-  if (!name.empty())
-    m_name.AddString(langCode, name);
+  if (name.empty())
+    return;
+
+  ASSERT_NOT_EQUAL(StringUtf8Multilang::kDefaultCode, langCode,
+                   ("Direct editing of default name is deprecated."));
+
+  if (!Editor::Instance().WasDefaultNameSaved(GetID()))
+  {
+    const auto mwmInfo = GetID().m_mwmId.GetInfo();
+
+    if (mwmInfo)
+    {
+      vector<int8_t> mwmLanguages;
+      mwmInfo->GetRegionData().GetLanguages(mwmLanguages);
+
+      if (CanUseAsDefaultName(langCode, m_name, mwmLanguages))
+        m_name.AddString(StringUtf8Multilang::kDefaultCode, name);
+    }
+  }
+
+  m_name.AddString(langCode, name);
+}
+
+// static
+bool EditableMapObject::CanUseAsDefaultName(int8_t const langCode, StringUtf8Multilang const & name,
+                                            vector<int8_t> const & nativeMwmLanguages)
+{
+  for (auto const & nativeLangCode : nativeMwmLanguages)
+  {
+    if (StringUtf8Multilang::kUnsupportedLanguageCode == nativeLangCode)
```

Comment:
In any case, why not to just skip this language, and continue? Why do you return false?